### ISSUE TITLE
has_one with sub class does not work

### ISSUE THREAD

**Issue by kidlab** [hzamani/active_record-acts_as#9]
Given these models:

``` ruby
class Photo < ActiveRecord::Base
  actable
  belongs_to :user
end

class Avatar < ActiveRecord::Base
  acts_as :photo
end

class User < ActiveRecord::Base
  has_one :avatar
  has_many :photos
end
```

When I call `user.avatar` it raises an error like this:

```
ActiveRecord::StatementInvalid: PG::UndefinedColumn: ERROR:  column avatars.user_id does not exist
```

It seems that we haven't supported this feature yet. Is there anyway we can work around this? 
Thanks.

**Comment by hachpai** [comment]
I've the same issue. The query doesn't inherit correctly of the acting_as model, so column name is wrong.

**Comment by icemall** [comment]
same problem

**Comment by manuelmeurer** [comment]
@hachpai @icemall Could you post your code and db schema?

**Comment by manuelmeurer** [comment]
This problem is probably fixed in v2.0.3 which was just released. Please open another issue if it still happens with that version.

**Comment by amrdruid** [comment]
Works fine in the new release
